frontend/lib/src/components/shared/StreamlitMarkdown/StreamlitMarkdown.tsx

```diff
@@ -379,41 +381,88 @@ export function RenderedMarkdown({
-      visit(tree, node => {
-        if (node.type === "textDirective") {
-          const nodeName = String(node.name)
-          if (colorMapping.has(nodeName)) {
-            const data = node.data || (node.data = {})
-            const style = colorMapping.get(nodeName)
-            data.hName = "span"
-            data.hProperties = data.hProperties || {}
-            data.hProperties.style = style
-            // Add class for background color for custom styling
-            if (
-              style &&
-              (/background-color:/.test(style) || /background:/.test(style))
-            ) {
-              data.hProperties.className =
-                (data.hProperties.className || "") + " has-background-color"
-            }
+      visit(tree, "textDirective", (node, _index, _parent) => {
+        const nodeName = String(node.name)
+        if (colorMapping.has(nodeName)) {
+          const data = node.data || (node.data = {})
+          const style = colorMapping.get(nodeName)
+          data.hName = "span"
+          data.hProperties = data.hProperties || {}
+          data.hProperties.style = style
+          // Add class for background color for custom styling
+          if (
+            style &&
+            (/background-color:/.test(style) || /background:/.test(style))
+          ) {
+            data.hProperties.className =
+              (data.hProperties.className || "") + " has-background-color"
           }
+        } else {
+          // Workaround to convert unsupported text directives to plain text to avoid them being
+          // ignored / not rendered. See https://github.com/streamlit/streamlit/issues/8726,
+          // https://github.com/streamlit/streamlit/issues/5968
+          node.type = "text"
+          node.value = `:${nodeName}`
+          node.data = {}
         }
       })
     }
   }
+
+  function remarkMaterialIcons() {
+    return (tree: any) => {
+      function replace(fullMatch: string, iconName: string): any {
+        return {
+          type: "text",
+          value: fullMatch,
+          data: {
+            hName: "span",
+            hProperties: {
+              role: "img",
+              ariaLabel: iconName + " icon",
+              style: {
+                display: "inline-block",
+                fontFamily: "Material Symbols Rounded",
+                // Disable selection for copying it as text.
+                // Allowing this leads to copying the underlying icon name,
+                // which can be confusing / unexpected.
+                userSelect: "none",
+                verticalAlign: "bottom",
+                fontWeight: "normal",
+                whiteSpace: "nowrap",
+                wordWrap: "normal",
+              },
+            },
+            hChildren: [{ type: "text", value: iconName }],
+          },
+        }
+      }
+      // We replace all `:material/` occurrences with `:material_` to avoid
+      // conflicts with the directive plugin.
```

Comment:
imho this comment slightly does not correspond to what is actually happening in this place. We could say that here `:material/` is already replaced by `:material_`, that's why we substitute `span` via regexp with `:material_`

Reply:
That makes sense, I updated the comment 👍